commit 1e295ad72d7c31da8cc5ff8a96082188b55e7298 (refs/changes/12/1212/3)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2017-01-06T23:48:30+00:00 (2 years, 9 months ago)
    
    MORAY-386 Document using Moray in standalone mode
    MORAY-387 Remove Moray's "microtime" dependency
    Reviewed by: David Pacheco <dap@joyent.com>
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Approved by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/CONTRIBUTING.md b/CONTRIBUTING.md
new file mode 100644
index 0000000..d67611f
--- /dev/null
+++ b/CONTRIBUTING.md
@@ -0,0 +1,19 @@
+# Contributing
+
+This repository uses [cr.joyent.us](https://cr.joyent.us) (Gerrit) for new
+changes. Anyone can submit changes. To get started, see the [cr.joyent.us user
+guide](https://github.com/joyent/joyent-gerrit/blob/master/docs/user/README.md).
+This repo does not use GitHub pull requests.
+
+See the [Joyent Engineering
+Guidelines](https://github.com/joyent/eng/blob/master/docs/index.md) for general
+best practices expected in this repository.
+
+Contributions should be "make prepush" clean. The "prepush" target runs the
+"check" target, which requires these separate tools:
+
+* https://github.com/davepacheco/jsstyle
+* https://github.com/davepacheco/javascriptlint
+
+If you're changing something non-trivial or user-facing, you may want to submit
+an issue first.
diff --git a/README.md b/README.md
index 94531de..310b6e0 100644
--- a/README.md
+++ b/README.md
@@ -1,12 +1,22 @@
+<!--
+    This Source Code Form is subject to the terms of the Mozilla Public
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, You can obtain one at http://mozilla.org/MPL/2.0/.
+-->
+
+<!--
+    Copyright 2017, Joyent, Inc.
+-->
+
 # Moray, the highly-available key/value store
 
 This repo contains Moray, the highly-available key/value store from Joyent.
-Moray offers a simple put/get/del (as well as search) protocol on top of
-Postgres 9.x, over plain TCP see <https://github.com/joyent/node-fast>.
+Moray offers a simple put/get/del/search protocol on top of Postgres 9.x,
+using the [node-fast](https://github.com/joyent/node-fast) RPC protocol.
 
-This repository is part of the Joyent SmartDataCenter project (SDC), and the
-Joyent Manta project.  For contribution guidelines, issues, and general
-documentation, visit the main [SDC](http://github.com/joyent/sdc) and
+This repository is part of the Joyent Manta and Triton projects. For
+contribution guidelines, issues, and general documentation, visit the main
+[Triton](http://github.com/joyent/triton) and
 [Manta](http://github.com/joyent/manta) project pages.
 
 ## Development
@@ -60,6 +70,27 @@ additional `moray-test` instance listening at port `2222`. Just scping into GZ
 and executing it should work.  You will need to configure the test suite
 appropriately (see the README.md in the moray-test-suite repository).
 
+## Running in standalone mode
+
+Moray can be used as a library to run a standalone Moray server that talks to a
+Postgres database without using Manatee. A single function is exported,
+`createServer`, which takes an object with the following fields:
+
+- `log`, a [bunyan](https://github.com/trentm/node-bunyan) logger
+- `port`, the TCP port to listen on
+- `bindip`, the IP address to bind to
+- `audit`, a boolean indicating whether to log the result and duration of all
+  requests
+- `standalone`, an object specifying the standalone server's configuration:
+    * `pg`, an object which specifies the Postgres client pool confguration:
+        - `queryTimeout`, how long (in milliseconds) before a query is timed out
+        - `maxConnections`, the maximum number of connections to maintain
+          to Postgres
+    * `url`, a [pg](https://github.com/brianc/node-postgres) URL describing how
+      to connect to the server (i.e., `/path/to/unix/socket/dir dbName`)
+
+`createServer` returns a server object with a `listen()` method to start the
+server. The server will emit a `ready` event once it's started up.
 
 ## License
 
diff --git a/deps/javascriptlint b/deps/javascriptlint
index e1bd0ab..ad52812 160000
--- a/deps/javascriptlint
+++ b/deps/javascriptlint
@@ -1 +1 @@
-Subproject commit e1bd0abfd424811af469d1ece3af131d95443924
+Subproject commit ad52812e77bdfb1e90fb71a1201adb2b665a27e6
diff --git a/deps/jsstyle b/deps/jsstyle
index da42b50..9600c7e 160000
--- a/deps/jsstyle
+++ b/deps/jsstyle
@@ -1 +1 @@
-Subproject commit da42b50ceb12d431437b32efd4c411a8e2fac0c8
+Subproject commit 9600c7e56c84f3a74d6e3d70c336e86f7a3e3769
diff --git a/lib/index.js b/lib/index.js
index bb39aba..0a36eee 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -15,9 +15,5 @@ var server = require('./server');
 ///--- Exports
 
 module.exports = {
-
-    createServer: server.createServer,
-    buckets: require('./buckets'),
-    objects: require('./objects')
-
+    createServer: server.createServer
 };
diff --git a/lib/manatee.js b/lib/manatee.js
index 8881e89..81b729d 100644
--- a/lib/manatee.js
+++ b/lib/manatee.js
@@ -12,7 +12,6 @@ var EventEmitter = require('events').EventEmitter;
 var util = require('util');
 
 var assert = require('assert-plus');
-var backoff = require('backoff');
 var bunyan = require('bunyan');
 var manatee = require('node-manatee');
 var once = require('once');
diff --git a/lib/objects/put.js b/lib/objects/put.js
index 781ab54..65742f7 100644
--- a/lib/objects/put.js
+++ b/lib/objects/put.js
@@ -11,7 +11,6 @@
 var util = require('util');
 
 var assert = require('assert-plus');
-var microtime = require('microtime');
 var crc = require('crc');
 var vasync = require('vasync');
 
@@ -49,7 +48,7 @@ function _createEtag(req) {
 
 
 function _now() {
-    return (Math.round((microtime.now()/1000)));
+    return (Date.now());
 }
 
 
diff --git a/lib/standalone.js b/lib/standalone.js
index 4ca8a97..aa0ef47 100644
--- a/lib/standalone.js
+++ b/lib/standalone.js
@@ -4,7 +4,6 @@ var EventEmitter = require('events').EventEmitter;
 var util = require('util');
 
 var assert = require('assert-plus');
-var backoff = require('backoff');
 var bunyan = require('bunyan');
 var once = require('once');
 var VError = require('verror').VError;
diff --git a/package.json b/package.json
index 351b210..e3d6eae 100644
--- a/package.json
+++ b/package.json
@@ -4,9 +4,9 @@
     "version": "2.1.0",
     "author": "Joyent (joyent.com)",
     "private": true,
+    "main": "lib/index.js",
     "dependencies": {
         "assert-plus": "0.1.5",
-        "backoff": "2.3.0",
         "bunyan": "0.22.1",
         "bunyan-syslog": "0.2.2",
         "clone": "0.1.11",
@@ -21,7 +21,6 @@
         "node-manatee": "git+https://github.com/joyent/node-manatee.git#152d3e1",
         "manatee": "git+https://github.com/joyent/manatee.git#8615736",
         "moray": "git+https://github.com/joyent/node-moray.git#fd5781b",
-        "microtime": "0.5.1",
         "once": "1.3.0",
         "panic": "0.2.1",
         "pg": "2.11.0",
